clicommand/agent_start.go

```diff
@@ -1257,6 +1270,28 @@ var AgentStartCommand = cli.Command{
 	},
 }
 
+// checkBinaryPaths looks up both the bootstrap and host buildkite-agent paths,
+// and returns an error if they do not match or if either cannot be determined.
+func checkBinaryPaths() error {
+	bootstrapPath, err := exec.LookPath("buildkite-agent")
```

Comment:
We may also want to resolve this to ensure it's not a symlink, to ensure we're comparing like for like